python/tests/plotting/plot_2d_test.py

```diff
@@ -239,7 +239,21 @@ def test_plot_2d_with_decreasing_edges():
 
 
 def test_plot_2d_binned_data():
-    plot(make_binned_data_array(ndim=2))
+    da = make_binned_data_array(ndim=2)
+    plot(da)
+    # Try without event-coord so implementation cannot use `histogram`
+    del da.bins.coords['yy']
+    da.coords['yy'] = da.coords['yy']['yy', 1:]
+
+
+def test_plot_2d_binned_data_non_counts():
+    da = make_binned_data_array(ndim=2)
+    da.events.unit = 'K'
+    plot(da)
```

Comment:
When trying this test in a notebook, it now works, but zooming in is very sluggish...
I don't know if that is expected or something new.

Reply:
I think the reason is the we have to fall back to binning and then using `bins.mean()`. Once we have an alternative to `histogram()` that can compute the mean this should be fixed. See TODO in code.